browser/app/js/components/Browse.js

```diff
@@ -480,338 +486,309 @@ export default class Browse extends React.Component {
-              okText='Delete'
-              cancelText='Cancel'
-              okHandler={ this.removeObject.bind(this) }
-              cancelHandler={ this.hideDeleteConfirmation.bind(this) }>
-            </ConfirmModal>
-            <Modal show={ shareObject.show }
-              animation={ false }
-              onHide={ this.hideShareObjectModal.bind(this) }
-              bsSize="small">
-              <ModalHeader>
-                Share Object
-              </ModalHeader>
-              <ModalBody>
-                <div className="input-group copy-text">
-                  <label>
-                    Shareable Link
-                  </label>
-                  <input type="text"
-                    ref="copyTextInput"
-                    readOnly="readOnly"
-                    value={ window.location.protocol + '//' + shareObject.url }
-                    onClick={ this.selectTexts.bind(this) } />
-                </div>
-                <div className="input-group" style={ { display: web.LoggedIn() ? 'block' : 'none' } }>
-                  <label>
-                    Expires in (Max 7 days)
-                  </label>
-                  <div className="set-expire">
-                    <div className="set-expire-item">
-                      <i className="set-expire-increase" onClick={ this.handleExpireValue.bind(this, 'expireDays', 1, shareObject.object) } />
-                      <div className="set-expire-title">
-                        Days
-                      </div>
-                      <div className="set-expire-value">
-                        <input ref="expireDays"
-                          type="number"
-                          min={ 0 }
-                          max={ 7 }
-                          defaultValue={ 5 }
-                          readOnly="readOnly"
-                        />
-                      </div>
-                      <i className="set-expire-decrease" onClick={ this.handleExpireValue.bind(this, 'expireDays', -1, shareObject.object) } />
+            </div>
+          </Modal>
+
+          <Modal className="policy" animation={ false } show={ showBucketPolicy } onHide={ this.hideBucketPolicy.bind (this) }>
+            <ModalHeader>
+              Bucket Policy
+              <small className="modal-header__sub">({ currentBucket })</small>
+              <i className="close close--dark" onClick={ this.hideBucketPolicy.bind (this) }>&times;</i>
+            </ModalHeader>
+
+            <div className="policy__body">
+              <PolicyInput bucket={ currentBucket }/>
+              { policies.map ((policy, i) => <Policy key={ i } prefix={ policy.prefix } policy={ policy.policy }/>) }
+            </div>
+          </Modal>
+
+          <ConfirmModal show={ deleteConfirmation.show }
+                        icon={ 'zmdi-alert-polygon c-red' }
+                        text='Are you sure you want to delete?'
+                        sub='This cannot be undone!'
```

Comment:
Good one! We can discuss this (and anything functionality related) with AB and others.